enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -1121,7 +1121,77 @@ func (s *SchedulerServer) GetSharedExecutorPoolGroupID() string {
 	return *sharedExecutorPoolGroupID
 }
 
-func (s *SchedulerServer) GetPoolInfo(ctx context.Context, os, requestedPool, workflowID string, poolType interfaces.PoolType) (*interfaces.PoolInfo, error) {
+func (s *SchedulerServer) getPoolOverrideFromExperiments(ctx context.Context, os, arch string, originalPool *interfaces.PoolInfo) *interfaces.PoolInfo {
+	fp := s.env.GetExperimentFlagProvider()
+	if fp == nil {
+		return nil
+	}
+
+	const experimentName = "remote_execution.pool_override"
+	obj := fp.Object(
+		ctx, experimentName, map[string]any{},
+		experiments.WithContext("os", os),
+		experiments.WithContext("arch", arch),
+		experiments.WithContext("pool", originalPool.Name),
+		experiments.WithContext("self_hosted", originalPool.IsSelfHosted),
+	)
+
+	// If null or empty, don't override.
+	if len(obj) == 0 {
+		return nil
+	}
+
+	log.CtxInfof(ctx, "Pool override: %+#v", obj)
+
+	override := &interfaces.PoolInfo{}
+	if groupIDValue, ok := obj["group_id"]; ok {
+		if groupID, ok := groupIDValue.(string); ok {
+			override.GroupID = groupID
+		} else {
+			log.CtxWarningf(ctx, "%s: 'group_id' is not a string: %v", experimentName, groupIDValue)
+			return nil
+		}
+	} else {
+		// group_id is not set; use the original pool's group ID.
+		override.GroupID = originalPool.GroupID
+	}
+	if poolValue, ok := obj["pool"]; ok {
+		if pool, ok := poolValue.(string); ok {
+			override.Name = pool
+		} else {
+			log.CtxWarningf(ctx, "%s: 'pool' is not a string: %v", experimentName, poolValue)
+			return nil
+		}
+	} else {
+		// pool is not set; use the original pool's name.
+		override.Name = originalPool.Name
+	}
+
+	if override.GroupID == *sharedExecutorPoolGroupID {
+		override.IsShared = true
+	} else {
+		override.IsSelfHosted = true
+	}
```

Comment:
this logic might be cleaner if we initialized the override from the original pool?

override := &interfaces.PoolInfo{}
copy(override, originalPool)

if ok := blah; blah {
  override.Blah = blah
}

etc

Reply:
yeah that makes more sense - done